clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1256,7 +1256,9 @@ private CompletedFetch initializeCompletedFetch(CompletedFetch nextCompletedFetc
                         log.debug("Discarding stale fetch response for partition {} since the fetched offset {} " +
                                 "does not match the current offset {}", tp, fetchOffset, subscriptions.position(tp));
                     } else if (subscriptions.hasDefaultOffsetResetPolicy()) {
-                        log.info("Fetch offset {} is out of range for partition {}, resetting offset", fetchOffset, tp);
+                        log.info("Fetch offset {} is out of range for partition {}. We only have log segments in " +
```

Comment:
The client may or may not be dealing with a broker version which supports returning the log start offset in the case of an out of range error. We should check whether the value is greater than 0 before including this information in the log message.

Reply:
Refactored